rust/kernel/allocator.rs

```diff
@@ -23,8 +24,42 @@ unsafe impl GlobalAlloc for KernelAllocator {
     }
 }
 
+unsafe impl<const A: bindings::gfp_t> Allocator for KernelAllocator<A> {
+    fn allocate(&self, layout: Layout) -> Result<NonNull<[u8]>, AllocError> {
+        // `krealloc()` is used instead of `kmalloc()` because the latter is
+        // an inline function and cannot be bound to as a result.
+        let mem = unsafe { bindings::krealloc(ptr::null(), layout.size(), A) as *mut u8 };
+        if mem.is_null() {
+            return Err(AllocError);
+        }
+        let mem = unsafe { core::slice::from_raw_parts_mut(mem, layout.size()) };
```

Comment:
We could use `ksize(mem)` to determine the actual size allocated and return that. Not sure whether it's worth it though.

Reply:
If `ksize` is very cheap, that is likely to be worth it. If it is not cheap, it is probably not worth it.

Reply:
`ksize` does some unpoision, not sure what that does.